From ca443c18acb28350f8018a9ffe68a308833f3fc0 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Sun, 28 Apr 2019 11:40:34 +0800
Subject: [PATCH] ota: Add i.mx8mq OTA support

This patch adds i.mx8mq OTA support according to the latest layout and state
machine diagram.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
---
 github.com/NXP/qoriq-edgescale-eds/startup/ota-statuscheck |  4 ++--
 github.com/NXP/qoriq-edgescale-eds/startup/ota-updateSet   | 15 +++++++++++----
 github.com/NXP/qoriq-edgescale-eds/startup/startup.sh      |  2 +-
 3 files changed, 14 insertions(+), 7 deletions(-)

diff --git a/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-statuscheck b/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-statuscheck
index e01521a..39e4ac5 100755
--- a/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-statuscheck
+++ b/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-statuscheck
@@ -6,7 +6,7 @@
 
 #!/bin/bash
 
-dd if=/dev/mmcblk0 of=/tmp/ota-info bs=512 skip=129024 count=1
+dd if=/dev/mmcblk1 of=/tmp/ota-info bs=512 skip=129024 count=1
 updateStatus=`cat /tmp/ota-info | awk '{print $1}'`
 platform=`cat /tmp/ota-info | awk '{print $2}'`
 oldimageVersion=`cat /tmp/ota-info | awk '{print $3}'`
@@ -33,5 +33,5 @@ fi
 updateStatus=0
 
 echo "$updateStatus $platform $oldimageVersion $newimageVersion $solutionname $solutionid $autoOta $mid" > /tmp/ota-info
-dd if=/tmp/ota-info of=/dev/mmcblk0 bs=512 seek=129024 count=1 conv=sync
+dd if=/tmp/ota-info of=/dev/mmcblk1 bs=512 seek=129024 count=1 conv=sync
 
diff --git a/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-updateSet b/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-updateSet
index c0608fc..b65ae38 100755
--- a/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-updateSet
+++ b/src/github.com/NXP/qoriq-edgescale-eds/startup/ota-updateSet
@@ -9,7 +9,7 @@
 /usr/local/edgescale/bin/cert-agent
 . /data/config.env
 
-dd if=/dev/mmcblk0 of=/tmp/ota-info bs=512 skip=129024 count=1
+dd if=/dev/mmcblk1 of=/tmp/ota-info bs=512 skip=129024 count=1
 platform=`cat /tmp/ota-info | awk '{print $2}'`
 oldimageVersion=`cat /tmp/ota-info | awk '{print $3}'`
 oldmid=`cat /tmp/ota-info | awk '{print $8}'`
@@ -41,13 +41,17 @@ get_imageURL(){
 
 download_images(){
 	feedback_status "ota-fetch"
-	download_path=/updateImages
+	download_path=/run/media/mmcblk1p1
+	mkdir -p /run/media
 	mkdir -p ${download_path}
+	mount /dev/mmcblk1p1 ${download_path}
+
+	mv ${download_path}/full_images.tgz ${download_path}/full_images.tgz.old
 
 	count=0
 	while [ $count -le 5 ]
 	do
-		wget -O /updateImages/full_images.tgz -c $imageURL
+		wget -O ${download_path}/full_images.tgz -c $imageURL
 		if [ $? != 0 ];then
 			((count++))
 			echo "retrying $count times"
@@ -77,10 +81,13 @@ if [ $imageURL ];then
 	feedback_status "ota-start"
 	updateStatus=8
 	echo "$updateStatus $platform $oldimageVersion $newimageVersion $solutionname $solutionid $autoOta $mid" > /tmp/ota-info
-	dd if=/tmp/ota-info of=/dev/mmcblk0 bs=512 seek=129024 count=1 conv=sync
+	dd if=/tmp/ota-info of=/dev/mmcblk1 bs=512 seek=129024 count=1 conv=sync
 
 	download_images
 	if [ $? -eq 0 ];then
+		updateStatus=4
+		echo "$updateStatus $platform $oldimageVersion $newimageVersion $solutionname $solutionid $autoOta $mid" > /tmp/ota-info
+		dd if=/tmp/ota-info of=/dev/mmcblk1 bs=512 seek=129024 count=1 conv=sync
 		sleep 5
 		reboot
 	else
diff --git a/git/src/github.com/NXP/qoriq-edgescale-eds/startup/startup.sh b/git/src/github.com/NXP/qoriq-edgescale-eds/startup/startup.sh
index 0e9edf7..daeb72c 100755
--- a/src/github.com/NXP/qoriq-edgescale-eds/startup/startup.sh
+++ b/src/github.com/NXP/qoriq-edgescale-eds/startup/startup.sh
@@ -25,7 +25,7 @@ mkdir -p /data
 
 ./env.sh
 start-stop-daemon --start --startas /bin/tee-supplicant --name tee-supplicant -m --pidfile /var/run/tee-supplicant.pid -b
-./cert-agent
+./cert-agent -dev /dev/mmcblk1
 
 . /data/config.env
 for env in $(set | grep ^ES)
-- 
2.7.4

